numpy/ctypeslib.py

```diff
@@ -401,8 +401,9 @@ def prep_pointer(pointer_obj, shape):
         attach an __array_interface__ property to it if it does not
         yet have one.
         """
-        try: pointer_obj.__array_interface__
+        try: pointer_obj.__array_interface__['shape'] = shape
         except AttributeError: pass
+        except TypeError: return
```

Comment:
Having separate error handlers is not a redundancy - it makes it explicit about which things are allowed to raise which errors. If `pointer_obj.__array_interface__` raises a `TypeError`, then presumably something is very wrong, and we should not silence it